doc/source/reference/c-api/array.rst

```diff
@@ -1711,6 +1711,17 @@ the functions that must be implemented for each slot.
    initial value is correct, since NumPy may call this even when it is not
    strictly necessary to do so.
 
+.. c:macro:: NPY_METH_static_data
+
+.. c:type:: void *static_data
+
+   Pointer to arbitrary data (known at compile time) that the method requires.
+   Typically used to pass in some data that allows the same loop to be used
+   for multiple methods or ufuncs. For instance, a bool could be used to
+   a string search function to indicate whether the search should be from
+   the from the front or the back, or a function pointer may be passed in
+   to do a specific test on the input.
```

Comment:
I added the new `static_data` - I hope this is the right place.

Note that this does *not* describe how to actually use the static data - as is, it is on the method object, which is opaque. Let me know what you think the best path forward is...

Reply:
OK, so with keeping the method private, this is no longer necessary.